src/taoensso/sente.cljc

```diff
@@ -930,125 +918,173 @@
-
-     (-chsk-send! [chsk ev opts]
-       (let [{?timeout-ms :timeout-ms ?cb :cb :keys [flush?]} opts
-             _ (assert-send-args ev ?timeout-ms ?cb)
-             ?cb-fn (cb-chan-as-fn ?cb ev)]
-         (if-not (:open? @state_) ; Definitely closed
-           (chsk-send->closed! ?cb-fn)
-
-           ;; TODO Buffer before sending (but honor `:flush?`)
-           (let [?cb-uuid (when ?cb-fn (enc/uuid-str 6))
-                 ppstr (pack packer (meta ev) ev ?cb-uuid)]
-
-             (when-let [cb-uuid ?cb-uuid]
-               (reset-in! cbs-waiting_ [cb-uuid] (have ?cb-fn))
-               (when-let [timeout-ms ?timeout-ms]
-                 (go
-                   (<! (async/timeout timeout-ms))
-                   (when-let [cb-fn* (pull-unused-cb-fn! cbs-waiting_ ?cb-uuid)]
-                     (cb-fn* :chsk/timeout)))))
-
-             (try
-               (.send @socket_ ppstr)
-               (reset! udt-last-comms_ (enc/now-udt))
-               :apparent-success
-               (catch :default e
-                 (errorf e "Chsk send error")
-                 (when-let [cb-uuid ?cb-uuid]
-                   (let [cb-fn* (or (pull-unused-cb-fn! cbs-waiting_ cb-uuid)
-                                    (have ?cb-fn))]
-                     (cb-fn* :chsk/error)))
-                 false))))))
 
-     (-chsk-connect! [chsk]
-       (when-let [WebSocket
-                  (or
-                    (enc/oget goog/global    "WebSocket")
-                    (enc/oget goog/global "MozWebSocket")
-                    (enc/oget @?node-npm-websocket_ "w3cwebsocket"))]
-
-         (let [instance-handle (reset! instance-handle_ (enc/uuid-str))
-               have-handle? (fn [] (= @instance-handle_ instance-handle))
-               connect-fn
-               (fn connect-fn []
-                 (when (have-handle?)
-                   (let [retry-fn
-                         (fn [] ; Backoff then recur
-                           (when (have-handle?)
-                             (let [retry-count* (swap! retry-count_ inc)
-                                   backoff-ms (backoff-ms-fn retry-count*)
-                                   udt-next-reconnect (+ (enc/now-udt) backoff-ms)]
-                               (warnf "Chsk is closed: will try reconnect attempt (%s) in %s ms"
+;; default JS client Chan
+#?(:cljs
+   (defrecord JSDefaultWebSocket [socket]
+     interfaces/IClientWebSocket     
+     (cws-close! [this]
+       (when socket
+         (.close socket 1000 "CLOSE_NORMAL")))
+     (cws-send! [this msg]
+       (when socket
+         (.send socket msg)))))
+
+#?(:cljs (defn js-websocket-creator [url {:keys [on-msg on-error on-close]}]
+           (when-let [Websocket (or
+                                 (enc/oget goog/global    "WebSocket")
+                                 (enc/oget goog/global "MozWebSocket")
+                                 (enc/oget @?node-npm-webclient-chan_ "w3cwebsocket"))]
```

Comment:
It looks like this has been renamed from `node-npm-websocket_`? I can't find where this comes from now though.